### PR TITLE
Fix missing attribute  '_from_state' in 'EmpiricalCovariance'

### PR DESCRIPTION
**Pull request by MarekWadinger**
Hey there,

although there is a support for initialization `_from_state()` in `proba.MultivariateGaussian`, it lacks implementation in `covariance.EmpiricalCovariance`.

What we should achieve, is the ability to define initial assumption about mean and covariance. Moreover, it might be practical to define covariance by single float as a small value to stabilize initial updates (problem related to Welford Algorithm)

```python
from river.proba import MultivariateGaussian

gaussian = MultivariateGaussian._from_state(  # type: ignore
    1, {"a": 0, "b": 0}, 1e-8, 1
)

gaussian = MultivariateGaussian._from_state(  # type: ignore
    1,
    {"a": 0, "b": 0},
    {("a", "a"): 1, ("a", "b"): 0, ("b", "a"): 0.5, ("b", "b"): 1},
    1,
)
```

Currently, following error is raised:
```
AttributeError: type object 'EmpiricalCovariance' has no attribute '_from_state'
```

In this pull request, I propose `_from_state()` implementation to `covariance.EmpiricalCovariance`, in accordance with implementation of `update_many()`

### CODE REVIEW

File: river/covariance/emp.py [hunk 1]
```diff
@@ -183,27 +183,65 @@ def update_many(self, X: pd.DataFrame):
             )
         }
 
-        for i, j in itertools.combinations(sorted(mean.keys()), r=2):
+        self = self._from_state(n=n, mean=mean, cov=cov, ddof=self.ddof)
```
[the remark this answers is not in the document]
Comment (MarekWadinger) [reply river/covariance/emp.py]: Thank you for explaining this. Do I understand correctly, that _from_state() method alone is  not a problem, but calling it in update_many function, possibly repeatedly, is?

If that's the case, please check if my latest commit [742aaf1](https://github.com/online-ml/river/pull/1471/commits/742aaf1436e3f996a60c15ba7b1733916d66464f) resolves the issue. :)

Looking forward to resolve this together!

**Comment by MarekWadinger** [comment]
Hello @MaxHalford,

this is only a gentle reminder of my [comment](https://github.com/online-ml/river/pull/1471#discussion_r1420076212) and commits that you might have missed. Would be happy to resolve this together or close it if the current solution would still violate the statefulness requirement.

> Thank you for explaining this. Do I understand correctly, that _from_state() method alone is not a problem, but calling it in update_many function, possibly repeatedly, is?
> 
> If that's the case, please check if my latest commit [742aaf1](https://github.com/online-ml/river/pull/1471/commits/742aaf1436e3f996a60c15ba7b1733916d66464f) resolves the issue. :)
> 
> Looking forward to resolve this together!

**Comment by MarekWadinger** [comment]
Hey there @MaxHalford and @smastelini, 👋

trying to gently remind you about this pull request once again since I believe having _from_state() is useful for model backup and recovery.

I believe I resolved @MaxHalford s concern about breaking statefulness. Could you please have a look at proposed changes? 👀

Thank you a lot. 🙏
Marek

File: river/covariance/emp.py [hunk 2]
```diff
@@ -183,27 +183,79 @@ def update_many(self, X: pd.DataFrame):
             )
         }
 
-        for i, j in itertools.combinations(sorted(mean.keys()), r=2):
+        self._update_from_state(n=n, mean=mean, cov=cov)
+
+    def _update_from_state(self, n: int, mean: dict, cov: float | dict):
```
Comment (smastelini) [inline river/covariance/emp.py:188]: Hi @MarekWadinger. Although `ddof` is listed in the docstring, it does not appear as a parameter.

Comment (MarekWadinger) [reply river/covariance/emp.py:188]: Hello @smastelini. Thank you for point out to this. It shouldn't be in the docstring. Entry was removed in the last commit [#40598](https://github.com/online-ml/river/pull/1471/commits/40598711f1b14ead274a9bc0cd476cd247510c0f)

File: river/covariance/emp.py [hunk 3]
```diff
@@ -183,27 +183,79 @@ def update_many(self, X: pd.DataFrame):
...
 
         for i in mean.keys():
             try:
                 self[i, i]
             except KeyError:
                 self._cov[i, i] = stats.Var(self.ddof)
-            self._cov[i, i] += stats.Var._from_state(
-                n=n, m=mean[i], sig=cov[i, i], ddof=self.ddof
-            )
+            if isinstance(cov, dict):
```
[the remark this answers is not in the document]
Comment (MarekWadinger) [reply river/covariance/emp.py:226]: Indeed, there was an error in the if statement, checking whether the cov is dict twice. Fixed the statement in the last commit [#40598](https://github.com/online-ml/river/pull/1471/commits/40598711f1b14ead274a9bc0cd476cd247510c0f)

**Comment by MarekWadinger** [comment]
@smastelini I fixed two more problems based on the test results. Now, everything should be sound. 😌

**Comment by smastelini** [comment]
It looks good to me now, but I will leave the final word to @MaxHalford once he has some spare time. :D

**Comment by MaxHalford** [comment]
Looks good to me, let's merge :)

**Pull request merged**